src/map.c

```diff
@@ -2156,8 +2157,6 @@ put_escstr(FILE *fd, char_u *strstart, int what)
 
     for ( ; *str != NUL; ++str)
     {
-	char_u	*p;
```

Comment:
Why is this declaration moved?

Reply:
Basically, following the C convention of declaring all variables on the top of the function.
I wondered why `p` was an exception. My educated guess was that because it was only used once.
Now is used in several places like the others.

Reply:
let me move it back. That change is unrelated and we should declare variables in the smallest block if possible. I'll also remove the braces around the single statement.